crates/deltalake-core/src/table/mod.rs

```diff
@@ -208,6 +220,21 @@ impl DeltaTableMetaData {
             })
             .collect()
     }
+
+    /// Return the column mapping mode according to delta.columnMapping.mode
+    pub fn get_column_mapping_mode(&self) -> Result<ColumnMappingMode, DeltaTableError> {
+        let mapping_mode = self.configuration.get("delta.columnMapping.mode");
+        match mapping_mode {
+            None => Ok(ColumnMappingMode::None),
+            Some(None) => Ok(ColumnMappingMode::None),
+            Some(Some(s)) => match s.as_str() {
+                "none" => Ok(ColumnMappingMode::None),
+                "id" => Ok(ColumnMappingMode::Id),
+                "name" => Ok(ColumnMappingMode::Name),
+                _ => Err(DeltaTableError::InvalidColumnMappingMode(s.clone())),
+            },
+        }
+    }
```

Comment:
Could we integrate this with the rest of the config in `crate::table::config`? After that we would ideally be reading that config via `state.table_config().column_mapping_mode()`

Reply:
I did that, and tried to align to code more with IsolationLevel/CheckpointPolicy
The error handling is a bit worse now, since the methods in table_config() always do unwrap_or_default instead of returning a Result, but I'd regard this as a separate topic.